committer-tools/refresh_collaborators.py

```diff
@@ -0,0 +1,143 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+This script automates the process of fetching contributor data from GitHub
+repositories, filtering top contributors who are not part of the existing
+committers, and updating a local configuration file (.asf.yaml) to include these
+new contributors.
+"""
+
+import io
+import logging
+import os
+from datetime import datetime, timedelta
+from typing import Dict, List, Tuple
+
+from bs4 import BeautifulSoup
+from github import Github
+from github.Commit import Commit
+from github.ContentFile import ContentFile
+from github.PaginatedList import PaginatedList
+from github.Repository import Repository
+from ruamel.yaml import YAML
```

Comment:
Is there a reason why we need this particular YAML parser instead of using the more common PyYAML?

Reply:
Yep, PyYAML does not handle well and removes comments. ruamel.yaml can handle them since they are inside collections as I have described at PR description (see paragraph about it)

Reply:
Thanks, I missed that 👍